libethereum/Client.cpp

```diff
@@ -119,9 +119,23 @@ Client::Client( ChainParams const& _params, int _networkID,
 }
 
 Client::~Client() {
+    stopWorking();
+}
+
+void Client::stopWorking() {
+    Worker::stopWorking();
+
     if ( m_snapshotHashComputing != nullptr )
         m_snapshotHashComputing->join();
 
+    if ( m_snapshotHashComputing != nullptr ) {
+        try {
+            if ( m_snapshotHashComputing->joinable() )
```

Comment:
We don't need here neither `if` no `try` - because the only other join() with this thread is done in Client::importTransactionsAsBlock, and that function is executed under mutex which is waited to be released in SkaleHost::stopWorking() (m_consensusWorkingMutex)

Reply:
debugging of many simiar joins shows it's needed